State of the Fin | 2026-05-24 (#1854)
jellyfin/jellyfin.org: blog/2026/05-22-state-of-the-fin/index.mdx

And of course, if you want to contribute code, we welcome it! Bug fixes are always welcome as-is; but features require a bit more work and a policy that we are still in the process of defining. But if you want to work on a feature, the short answer is: please ask us before you start implementing it and work with the team on a full design/scope discussion to help ensure your work will be accepted!

## Development Updates

<ActivityChart />

#### What's New

We've continued pushing bug and [security](https://github.com/jellyfin/jellyfin/security/advisories) fixes to the stable release. Following the 10.11.9 release, development attention has largely shifted toward finalizing 12.0 ahead of the upcoming release candidate phase. A few items still need to be addressed before we can ship RC1: [Jellyfin 12.0](https://github.com/orgs/jellyfin/projects/73)

> The [Performance PR](https://github.com/jellyfin/jellyfin/pull/16062) which forms the foundation for much of the work in 12.0, has been merged into master and is now part of the weekly unstable releases. These are published every Monday, though we may push additional out-of-band builds for a while to support testing. Unstable builds can be accessed via Docker using the "unstable" tag or by selecting the unstable tab on our download page.

Reviewer: ```suggestion
The [Performance PR](https://github.com/jellyfin/jellyfin/pull/16062) which forms the foundation for much of the work in 12.0, has been merged into master and is now part of the weekly unstable releases. These are published every Monday, though we may push additional out-of-band builds for a while to support testing. Unstable builds can be accessed via Docker using the "unstable" tag or by selecting the unstable tab on our downloads page.
```

*- [Joe Kribs](https://github.com/JPKribs)*

### Other

#### New Clients

Work has started on two new Jellyfin clients!

Currently in the prototype phase is a Jellyfin client for the new **Vega OS** platform from Amazon. This is the operating system used on the newer models of Fire TV Sticks released since 2025. We've seen a lot of requests for this as people buy the devices thinking they can use our Android TV app and end up not finding Jellyfin.

Further into development is our client for **Titan OS**. This smart TV platform is mainly found on Philips televisions. We're currently testing this client and hopefully have some news to share soon!

## Documentation

### [Jellyfin.org](https://github.com/jellyfin/jellyfin.org)

#### What's New

Documentation is always a work in progress but over this period Jellyfin's documentation has seen some notable changes:

- Improved CI to catch common issues and spelling errors
- A new feature proposal guideline
- Fish shell support added to the installer commands
- Cleanup of reverse proxy documentation
- The LLM policy made its way into the docs

> #### Moving Forward

Reviewer 2: Moving Forward sounds like "whats planned"
I do have some greater plans regarding documentation restructure.
Since almost a full year we have stopped reorganizing individual guides to keep it easier to find things.
The idea of the restructure is to fully reorganize all existing docs AT ONCE so that its ONE event where a lot will change that we can officially announce.
https://github.com/jellyfin/jellyfin-meta/discussions/111

Author: Would you be able to provide the changes that you would like included in this section? Sorry, I haven't been too in-the-know for documentation so I don't want to misrepresent anything ya'll are up to haha

Author: @BotBlake I think we're in the last ~15hr or so before I flip this for go live. I wanted to check if you had a change or revision to what was written out here before that point. Let me know or feel free to leave a revision!

This PR addresses a number of slow queries in 10.11.x, particularly for folder-based libraries like home videos & photos, and books. Other areas have also seen improvements, including collections, music libraries, playlists, and homepage load times. For more details, see the PR description.

12.0 also includes a fix for EPG issues with the Schedules Direct API, as well as a fix for errors when restructuring media libraries. A long-requested feature, support for multiple versions of episodes in a series library, has also been added. While users may notice some incidental improvements in scanner performance as a result of other changes, scanner optimization itself was not a primary focus of this release. More substantial work in that area is planned for 13.0, which is currently in the early [planning phase](https://github.com/jellyfin/jellyfin-meta/discussions/125).

> #### Testing

Author: Server changes needed here for 12.0 & recent PRs.

And of course, if you want to contribute code, we welcome it! Bug fixes are always welcome as-is; but features require a bit more work and a policy that we are still in the process of defining. But if you want to work on a feature, the short answer is: please ask us before you start implementing it and work with the team on a full design/scope discussion to help ensure your work will be accepted!

## Development Updates

<ActivityChart />

#### What's New

We've continued pushing bug and [security](https://github.com/jellyfin/jellyfin/security/advisories) fixes to the stable release. The upcoming 10.11.9 release is expected to be the final release in that branch, and we're now focused on 12.0, with the first release candidate planned in the coming weeks. A few items still need to be addressed before we can ship RC1: [Jellyfin 12.0](https://github.com/orgs/jellyfin/projects/73)

> For those following GitHub, the [Performance PR](https://github.com/jellyfin/jellyfin/pull/16062) has been merged into master and is available for testing in the weekly unstable releases. These are published every Monday, though we may push additional out-of-band builds for a while to support testing. Unstable builds can be accessed via Docker using the "unstable" tag or by selecting the unstable tab on our download page.

Reviewer: ```suggestion
We've continued pushing bug and [security](https://github.com/jellyfin/jellyfin/security/advisories) fixes to the stable release. Following the 10.11.9 release, development attention has largely shifted toward finalizing 12.0 ahead of the upcoming release candidate phase. A few items still need to be addressed before we can ship RC1: [Jellyfin 12.0](https://github.com/orgs/jellyfin/projects/73)

The [Performance PR](https://github.com/jellyfin/jellyfin/pull/16062) which forms the foundation for much of the work in 12.0, has been merged into master and is now part of the weekly unstable releases. These are published every Monday, though we may push additional out-of-band builds for a while to support testing. Unstable builds can be accessed via Docker using the "unstable" tag or by selecting the unstable tab on our download page.
```

Before testing the unstable 12.0 build, please note the following:

Unstable builds are not guaranteed to work from one build to another. These builds should only be used for testing or evaluating new features, and running them as a daily driver is at your own risk.

If upgrading from an earlier version, a full backup of the data directory is strongly recommended, as this release includes database changes that prevent rolling back without a full restore.

Installed repository plugins (anything not built-in) should also be removed before migrating. Plugins will likely need time to adapt to the new database changes, so re-adding them afterward is the safest approach for testing.

Official plugins compatible with Jellyfin are available through the unstable plugin repository. To test plugins for 12.0:

- Go to Dashboard -> Plugins -> Manage Repositories
- Update the Plugin Repository URL to: https://repo.jellyfin.org/files/plugin-unstable/manifest.json

After migrating:

- Perform a full library scan to restore alternative versions
- Run the "Optimize database" scheduled task

If you run into issues, please prefix bug reports with "[12.0]".
#### Client Development Changes

> As mentioned [before](/blog/2025/10-19-jellyfin-release-10.11.0/index.mdx#client-development-changes), the API will no longer allow deprecated authorization mechanisms by default, so clients and tooling that haven't already migrated will need to update.

Reviewer: ```suggestion
As mentioned [before](/blog/2025/10-19-jellyfin-release-10.11.0/index.mdx#client-development-changes), the API will no longer allow deprecated authorization mechanisms by default, so clients and tooling that haven't already migrated will need to update. Any new API changes or breaking modifications will now be deferred to 13.0, allowing client developers to begin targeting a stable API surface ahead of release.
```

> We've also added an option for developers to test their clients with deprecated authorization methods fully disabled. More details on how to authorize and run these tests are available in this [document](https://gist.github.com/nielsvanvelzen/ea047d9028f676185832e51ffaf12a6f).

Reviewer: ```suggestion
As mentioned [before](/blog/2025/10-19-jellyfin-release-10.11.0/index.mdx#client-development-changes), the API will no longer allow deprecated authorization mechanisms by default, so clients and tooling that haven't already migrated will need to update. Any new API changes or breaking modifications not already in progress will now be deferred to 13.0, allowing client developers to begin targeting a stable API surface ahead of release.
```
